File: backend/src/main/java/com/niyotechnologies/claimlens/user/service/UserService.java
package com.niyotechnologies.claimlens.user.service;

import com.niyotechnologies.claimlens.user.dto.CreateUserRequest;
import com.niyotechnologies.claimlens.user.dto.UpdateUserRequest;
import com.niyotechnologies.claimlens.user.dto.UserResponse;
import com.niyotechnologies.claimlens.common.response.PagedResponse;

import java.util.List;

public interface UserService {

    /** Paged directory listing for the Users screen (all roles). */
    PagedResponse<UserResponse> list(int page, int size);

    /** All users (unpaged), optionally filtered to one role code (e.g. INVESTIGATOR) — for pickers. */
    List<UserResponse> options(String roleCode);

    UserResponse create(CreateUserRequest request);

    UserResponse update(Long userId, UpdateUserRequest request);
}

File: frontend/services/userService.ts
import { http } from "@/lib/http";
import type {
  ApiResponse,
  CreateUserRequest,
  PagedResponse,
  PageParams,
  UpdateUserRequest,
  UserResponse,
} from "@/lib/types";

const unwrap = <T>(p: Promise<{ data: ApiResponse<T> }>) => p.then((r) => r.data.data);

export const userService = {
  /** Paged directory listing for the Users screen. */
  list: (params: PageParams = {}) =>
    unwrap(http.get<ApiResponse<PagedResponse<UserResponse>>>("/users", { params })),

  /** All users (unpaged), optionally filtered by role code (e.g. "INVESTIGATOR") — for pickers. */
  options: (roleCode?: string) =>
    unwrap(
      http.get<ApiResponse<UserResponse[]>>("/users/options", {
        params: roleCode ? { role: roleCode } : undefined,
      }),
    ),

  create: (body: CreateUserRequest) =>
    unwrap(http.post<ApiResponse<UserResponse>>("/users", body)),

  update: (userId: number, body: UpdateUserRequest) =>
    unwrap(http.put<ApiResponse<UserResponse>>(`/users/${userId}`, body)),
};
